pgxn/neon/file_cache.c

```diff
@@ -100,7 +105,9 @@ typedef struct FileCacheEntry
 	BufferTag	key;
 	uint32		hash;
 	uint32		offset;
-	uint32		access_count;
+	uint32		access_count : 30;
+	uint32      prewarm_requested : 1; /* entry should be filled by prewarm */
+	uint32      prewarm_started : 1;   /* chunk is written by lfc_prewarm */
```

Comment:
Sorry, I do not completely understand your arguments here.
Yes, it is possible to move this bits to separate array, but why? 30 bits is more than enough for access_counter - there will be never milliard backbends. So this bits are just wasting. And size of this structure is quite critical - because it is multiplied by maximal possible LFC size. Certainly it is possible to make it more compact for example by replacing pointers with indexes for LRU list. But still I do not see any good reasons in replacing bit fields with bool.

Also I completely do not understand you concern about using 1-bit field as bool. It is quite common practice in C and in Postgres code also. I do not see anything dangerous, inconvenient tr confusing here.